knife/lib/chef/knife/bootstrap.rb

```diff
@@ -1188,6 +1187,24 @@ def set_transport_options(opts)
 
         connection&.connection&.transport_options&.merge! opts
       end
+
+      # List Train transports available (but not loaded) on system.
+      #
+      # Will filter out train-core, API transports, and other incompatible ones.
+      #
+      # @return [Array] transport_names which can be used for command execution
+      def transports_available
+        train_gems = Gem::Specification.select { |spec| spec.name.start_with? "train-" }
+        train_gems.delete_if { |spec| TRANSPORTS_BLACKLIST.include? spec.name }
+        train_gems.map { |spec| spec.name.delete_prefix "train-" }
+      end
```

Comment:
maybe to help debug here do:
```suggestion
      def transports_available
        train_gems = Gem::Specification.select { |spec| spec.name.start_with? "train-" }
        puts "DEBUG: #{train_gems}"
        train_gems.delete_if { |spec| TRANSPORTS_BLACKLIST.include? spec.name }
        train_gems.map { |spec| spec.name.delete_prefix "train-" }
      end
```